2016-10-28  Sahil Manoj Gandhi  <gandhi@lnxsrv07.seas.ucla.edu>

	* /lib/timespec.h Changed from int to long comparison to deal with overflow.

--- lib/timespec.h	2016-10-28 00:33:06.000000000 -0700
+++ lib/timespecTemp.txt	2016-10-28 11:30:23.000000000 -0700
@@ -45,8 +45,15 @@
 static inline int
 timespec_cmp (struct timespec a, struct timespec b)
 {
-  int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+
+  signed long long int difference = (signed long long int) a.tv_sec - (signed long long int) b.tv_sec;
+
+  if (difference > 0)
+    return 1;
+  else if (difference == 0)
+    return 0;
+  else
+    return -1;
 }
 
 # if ! HAVE_DECL_NANOSLEEP
